### ISSUE TITLE
Use Tags for Dependencies with `FetchContent_Declare`

### ISSUE THREAD

**Issue by dschiller** [CMakePP/CMakeTest#97]
**Is your feature request related to a problem? Please describe.**
You introduced a breaking change with https://github.com/CMakePP/CMakeTest/commit/393a85e774a2c4cb89ac720abe3ab252213bf933 for all your current depndencies (CMaize).

**Describe the solution you'd like**
For all your dependencies you also need to pin to tags with the `GIT_TAG` attribute (see https://cmake.org/cmake/help/latest/module/FetchContent.html) to not introduce breaklng changes.

**Describe alternatives you've considered**
Forking your CMakeTest repro and fix the `FetchContent_Declare` usage by using tags.

**Additional context**
It is good practice to pin to a tag instead to the master branch for dependencies.

**Comment by dschiller** [comment]
Yes Ryan, pinning the the version of CMaize and all future Dependencies to a specific tag.

**Comment by dschiller** [comment]
Regarding our broken code we believe that you changed the Cpp repository from public to private which made our code fail as we didn't update to use CMaize that time you switched to CMaize.

**Comment by ryanmrichard** [comment]
@dschiller just to clarify, CMaize is Cpp. We renamed it. Nonetheless, #98 pins CMakeTest's dependencies so hopefully this issue won't come up again.

**Comment by dschiller** [comment]
Okay, thank you for the clarification Ryan. And also thank you a lot for this fast solution!
